### ISSUE TITLE
Weird whitespace with JSX Fragment & Typescript in VS Code

### ISSUE THREAD

**Issue by bazo** [prettier/prettier-vscode#651]
Hi,

i've been using create react app and prettier in vscode for a while. code was in plain js and jsx. now i want to migrate files to tsx and found a weird formatting issue when top level element in render in a fragment <>

attaching animated gif to explain. 
![div to fragment](https://user-images.githubusercontent.com/181588/48845071-e0adcb00-ed9b-11e8-8345-cf28b78f7f42.gif)

this only happens when the file has .tsx extension. 

and everytime i press ctrl+s or alt+shift+f it adds new \\" to those lines

**Comment by marcofugaro** [comment]
This happens to me as well! It's really annoying!

**Comment by alex996** [comment]
Same thing here, sigh 😞

**Comment by ntotten** [comment]
Please provide an example repository with repro steps.

**Comment by alex996** [comment]
@ntotten Here's [one with React](https://github.com/alex996/prettier-whitespace-bug-repro). Be sure to have ESLint and Prettier extensions installed. Then, to reproduce:

1. `git clone https://github.com/alex996/prettier-whitespace-bug-repro.git repro`
2. `cd repro`
3. `npm install`
4. `code .`
5. In `src/index.tsx`, change `<Fragment>` to `<>`
6. `Ctrl + S`

The code goes from

```tsx
const App = () => (
  <Fragment>
    <h1>Hello</h1>
  </Fragment>
)
```

to

```tsx
const App = () => (
  <>'   '<h1>Hello</h1>' '</>
)
```

In this case, it seems to be related to `standard`. If you go to `.eslintrc.json` and remove `"standard"` from `"extends"` array, the weird formatting goes away. However, if you don't set `"editor.formatOnSave": true`, then `npx eslint '**/*.tsx'` passes just fine either way.

**Comment by jamesonhill** [comment]
I have the same issue. Any .tsx file using the short hand fragment syntax inserts empty space between the opening tag & the first child opening tag as well as between the last closing child tag & the closing fragment tag.

**Comment by marcofugaro** [comment]
Hey! @alex996 did provide the repro! Please don't close this!

**Comment by bazo** [comment]
you got all information needed, why close this? 

i am now using this extension https://github.com/remimarsal/prettier-now-vscode
because it works correctly

**Comment by ntotten** [comment]
This should be fixed in version 3.0. If somebody has a chance please verify using the 3.0 preview of the extension: https://marketplace.visualstudio.com/items?itemName=Prettier.prettier-vscode
